File: Práctica2/java/src/deepspace/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */

// Representa los suministros de una estación espacial

public class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    public SuppliesPackage(float ammoP, float fuelU, float shieldP){
        this.ammoPower = ammoP;
        this.fuelUnits = fuelU;
        this.shieldPower = shieldP;        
    }

    public SuppliesPackage(SuppliesPackage s){
        this.ammoPower = s.ammoPower;
        this.fuelUnits = s.fuelUnits;
        this.shieldPower = s.shieldPower;
    }    
    
    float getAmmoPower(){
        return ammoPower;
    }
    
    float getFuelUnits(){
        return fuelUnits;
    }
    
    float getShieldPower(){
        return shieldPower;
    }
}

File: Práctica3/lib/SuppliesPackage.rb
#enconding: utf-8

# Representa los suministros para una estación espacial

module Deepspace
    class SuppliesPackage
        def initialize(ammopower, fuelunits, shieldpower)
            @ammoPower = ammopower
            @fuelUnits = fuelunits
            @shieldPower = shieldpower
        end

        def self.newCopy(package)   # Constructor de copia
            new(package.ammoPower, package.fuelUnits, package.shieldPower)
        end

        def ammoPower 
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end

        def shieldPower
            @shieldPower
        end
    end
end

# Código de prueba
# prueba = Deepspace::SuppliesPackage.new(1,1,1)
# puts prueba.fuelUnits
# pruebacopy = Deepspace::SuppliesPackage.new(2,2,2)
# puts pruebacopy.fuelUnits
# prueba2 = Deepspace::SuppliesPackage.newCopy(pruebacopy)
# puts prueba2.fuelUnits
